scrapi/harvesters/lwbin.py

```diff
@@ -0,0 +1,207 @@
+"""
+A Lake Winnipeg Basin Information Network (BIN) harvester for the SHARE project
+
+Example API request: http://130.179.67.140/api/3/action/package_search?q= (problematic)
+http://130.179.67.140/api/3/action/current_package_list_with_resources (currently using)
+It oddly returns 5 more datasets than all searchable ones on LWBIN data hub.
+
+Known issues:
+1 -- Five datasets can be searched but cannot be accessed via LWBIN.
+Clicking on the searching result would result in linking to a redirected page like this:
+http://130.179.67.140/user/login?came_from=http://130.179.67.140/dataset/mpca-surface-water-data-access-interactive-map
+Within each dataset there are resouces that contain urls to source pages. For future work considering using resources
+urls as canonical urls.
+2 -- Resouces properties contained in raw metadata of the datasets are not added to the normalized metadata at this
+point.
+3 -- Single name contributors can be used as filters or an invalid query will be returned. Has nothing to do with scrapi but the frontend.
+"""
+
+from __future__ import unicode_literals
+
+import json
+import logging
+
+from nameparser import HumanName
+from dateutil.parser import parse
+
+from scrapi import requests
+from scrapi.base import JSONHarvester
+from scrapi.linter.document import RawDocument
+from scrapi.base.helpers import build_properties
+
+
+logger = logging.getLogger(__name__)
+
+organization_dict = [
+    "organization", "Organization", "fund", "Fund", "canada", "Canada", "agriculture", "Agriculture", "commitee",
+    "Commitee", "international", "International", "council", "Council", "office", "Office", "of", "Of"
+]
+
+customized_organization_dict = ["LWBIN", "CoCoRaHS", "USGS"]
+
+
+def is_organization(name):
+    """Return a boolean to indicate if the name passed to the function is an organization
+    """
+    words = name.split(' ')
+    while len(words) > 0:
+        word = words.pop()
+        if word in organization_dict or word in customized_organization_dict:
+            return True
+
+
+def process_contributors(authors, emails):
```

Comment:
you should break this function up, it is too long. Try separating the logic for processing people and organizations into their own functions.